### PR TITLE
Lyft sidecar container backport / COMPUTE-307

### PR DESCRIPTION
**Pull request by EricMountain**
Lyft sidecar container backport.

See https://datadoghq.atlassian.net/browse/COMPUTE-307 for the commit list.

### CODE REVIEW

**Comment by lbernail** [comment]
The patch is far too big to really review, so I'd say let's test it seriously on fury?

File: pkg/kubelet/kuberuntime/kuberuntime_container_test.go [hunk 1]
```diff
@@ -293,7 +293,7 @@ func TestLifeCycleHook(t *testing.T) {
 		testPod.DeletionGracePeriodSeconds = &gracePeriodLocal
 		testPod.Spec.TerminationGracePeriodSeconds = &gracePeriodLocal
 
-		m.killContainer(testPod, cID, "foo", "testKill", &gracePeriodLocal)
+		m.killContainer(testPod, cID, "foo", "testKill", time.Duration(gracePeriodLocal)*time.Second)
```
Comment (nyodas) [inline pkg/kubelet/kuberuntime/kuberuntime_container_test.go]: Every time.Duration feels weird we never use the `time` itself , we cast it in int64 
This work but it's weird

File: pkg/kubelet/status/status_manager.go [hunk 2]
```diff
@@ -613,3 +613,78 @@ func normalizeStatus(pod *v1.Pod, status *v1.PodStatus) *v1.PodStatus {
 	kubetypes.SortInitContainerStatuses(pod, status.InitContainerStatuses)
 	return status
 }
+
+// mergePodStatus merges oldPodStatus and newPodStatus where pod conditions
```
Comment (nyodas) [inline pkg/kubelet/status/status_manager.go]: Is that because it's never called ??

**Comment by EricMountain** [comment]
> I think it need some real life test
> But overall i didn't see too much thing that appeared weird.

Right. The idea with the review is just to check there's nothing blatantly bad/stupid. I've taken the upstream commits as close to as-is as possible without questioning (much) the validity/reason of some changes (like the duration stuff).

And then of course it has to be tested... At least to make sure that sidecars don't break the rest. If sidecars have bugs, we'll have to fix those subsequently. We might be able to get some feedback from prod-platform now on the 1.14 release.

File: pkg/kubelet/status/status_manager.go (hunk 2 above)
Comment (EricMountain) [reply pkg/kubelet/status/status_manager.go]: It's code that is present in the release Lyft applied their changes on, but that is not part of their change. I should have deleted it, and will do now.

**Pull request merged**